docs/learn/bitesize-invaders.md

```diff
@@ -0,0 +1,309 @@
+```csharp
+/// <summary>
+///     UpdateAndCheckPlayersInLobby
+///     Checks to see if we have at least 2 or more people to start
+/// </summary>
+private void UpdateAndCheckPlayersInLobby()
+{
+    //This is game preference, but I am assuming at least 2 players?
+    m_AllPlayersInLobby = m_ClientsInLobby.Count > 1;
+
+    foreach (var clientLobbyStatus in m_ClientsInLobby)
+    {
+        SendClientReadyStatusUpdatesClientRpc(clientLobbyStatus.Key, clientLobbyStatus.Value);
+        if (!NetworkManager.Singleton.ConnectedClients.ContainsKey(clientLobbyStatus.Key))
+
+            //If some clients are still loading into the lobby scene then this is false
+             m_AllPlayersInLobby = false;
+    }
+
+    CheckForAllPlayersReady();
+}
+```
+
+<!-- ```csharp reference
+https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/LobbyControl.cs#L81-L100
+``` -->
+
+When all the players have joined the lobby and are ready, `UpdateAndCheckPlayersInLobby` calls `CheckForAllPlayersReady` to transition to the next scene.( lines 158186
+
+```csharp
+/// <summary>
+///     CheckForAllPlayersReady
+///     Checks to see if all players are ready, and if so launches the game
+/// </summary>
+private void CheckForAllPlayersReady()
+{
+    if (m_AllPlayersInLobby)
+    {
+        var allPlayersAreReady = true;
+        foreach (var clientLobbyStatus in m_ClientsInLobby)
+            if (!clientLobbyStatus.Value)
+
+                //If some clients are still loading into the lobby scene then this is false
+                allPlayersAreReady = false;
+
+        //Only if all players are ready
+        if (allPlayersAreReady)
+        {
+            //Remove our client connected callback
+            NetworkManager.Singleton.OnClientConnectedCallback -= OnClientConnectedCallback;
+
+            //Remove our scene loaded callback
+            SceneTransitionHandler.sceneTransitionHandler.OnClientLoadedScene -= ClientLoadedScene;
+
+            //Transition to the ingame scene
+            SceneTransitionHandler.sceneTransitionHandler.SwitchScene(m_InGameSceneName);
+        }
+    }
+}
+```
+
+<!-- ```csharp reference
+https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/LobbyControl.cs#L158-L186
+``` -->
+
+## Unconventional Networked Movement
+
+Invaders has an easy movement type - moving only on one (horizontal) axis - which allows you to only modify the transform client-side. Modify *[PlayerControl.cs](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/PlayerControl.cs)* in the `InGameUpdate` function with the help of a [`NetworkTransform`](../components/networktransform.md) that is attached directly to the Player Game Object and automatically syncs up the Transform with the other clients. It will smooth out by interpolating or extrapolating for all of them.
```

Comment:
Replace this with:

"Invaders has an easy movement type - moving only on one (horizontal) axis - which allows you to only modify the transform client-side without waiting for server-side validation. You can find where we perform the move logic in *[PlayerControl.cs](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/master/Basic/Invaders/Assets/Scripts/PlayerControl.cs)* in the `InGameUpdate` function. With the help of a [`NetworkTransform`](../components/networktransform.md) that is attached directly to the Player Game Object will automatically sync up the Transform with the other clients. At the same time, it will smooth out the movement by interpolating or extrapolating for all of them."